src/data/mv_polynomial/equiv.lean

```diff
@@ -319,6 +321,262 @@ begin
   simp only [mv_polynomial.fin_succ_equiv_apply, mv_polynomial.eval₂_hom_C],
 end
 
+lemma fin_succ_equiv_X_eq_zero {n : ℕ} {R : Type u} [comm_semiring R] :
+  fin_succ_equiv R n (X 0) = polynomial.X := by simp
+
+lemma fin_succ_equiv_X_ne_zero {n : ℕ} {R : Type u} [comm_semiring R] {j : fin n} :
+  fin_succ_equiv R n (X j.succ) = polynomial.C (X j) := by simp
+
+private lemma fin_succ_equiv_coeff_coeff_C {n : ℕ } {R : Type u} [comm_semiring R]
+  (a : R) (i : ℕ) (m : fin n →₀ ℕ) :
+  coeff m (((fin_succ_equiv R n) (C a)).coeff i) = coeff (finsupp.cons i m) (C a) :=
+begin
+  by_cases c_i : i = 0,
+  { simp only [c_i, fin_succ_equiv_apply, polynomial.coeff_C_zero, coeff_C, ring_hom.coe_comp,
+               eval₂_hom_C],
+    by_cases c_m : 0 = m,
+    { simp only [← c_m, if_true, eq_self_iff_true, finsupp.cons_zero_zero] },
+    { have x : ¬ finsupp.cons 0 m = 0,
+      { apply finsupp.cons_any_nonzero,
+        symmetry,
+        simpa using c_m },
+      have x' : ¬ 0 = finsupp.cons 0 m := by cc,
+      simp only [c_m, if_false, x', if_false] } },
+  { have x : ¬ finsupp.cons i m = 0,
+    { apply finsupp.cons_nonzero_any,
+      simpa using c_i },
+    have x' :  ¬ 0 = finsupp.cons i m := by cc,
+    simp only [fin_succ_equiv_apply, coeff_C, function.comp_app, ring_hom.coe_comp, eval₂_hom_C,
+               polynomial.coeff_C, c_i, if_false, coeff_zero, x', if_false] },
+end
+
+local attribute [instance] classical.prop_decidable -- TODO is it okay to use this here?
```

Comment:
It's better to use `classical` in the proof, or `letI := classical.dec foo` if `classical` causes trouble.